src/test/test-hashmap-plain.c

```diff
@@ -49,6 +49,35 @@ TEST(hashmap_replace) {
         ASSERT_STREQ(r, "val5");
 }
 
+TEST(hashmap_ensure_replace) {
+        _cleanup_hashmap_free_ Hashmap *m = NULL;
+        _cleanup_free_ char *val1 = NULL, *val2 = NULL;
+        char *s;
```

Comment:
I would get rid of this variable and just inline every call to hashmap_get() into the assertion